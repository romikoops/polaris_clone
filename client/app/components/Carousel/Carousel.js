import React, {Component} from 'react';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import styles from '../ActiveRoutes/ActiveRoutes.scss';
import {v4} from 'node-uuid';
export class Carousel extends Component {
    render() {
      if (!this.props.slides) {
        return '';
      }
      const slideNumber = this.props.noSlides ? this.props.noSlides : 4;
        const settings = {
            dots: true,
            autoplay: true,
            infinite: true,
            slidesToShow: slideNumber,
            speed: 2000,
            arrows: true
        };
        const slides = this.props.slides.map((route) => {
            const divStyle = {
                backgroundImage: 'url(' + route.image + ')'
            };
            return (
                <div key={v4()} className={styles.slick_slide + ' flex-none layout-row layout-align-center-center'} style={divStyle}>
                    {this.props.fade ? <div className={`flex-none ${styles.fade}`}></div> : ''}
                    <div className={`flex-none layout-column layout-align-center-center ${styles.slick_content}`}>
                        <h2 className={styles.slick_city + ' flex-none'}> {route.header} </h2>
                        <h5 className={styles.slick_country + ' flex-none'}> {route.subheader} </h5>
                    </div>
                </div>
            );
        });
        return (
            <div className={`flex-100 layout-row ${styles.slider_container}`}>
                <Slider {...settings}>
                    {slides}
                </Slider>
            </div>
        );
    }
}
