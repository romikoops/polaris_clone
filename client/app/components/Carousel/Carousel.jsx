import React from 'react'
import Slider from 'react-slick'
import { v4 } from 'node-uuid'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import 'slick-carousel/slick/slick.css'
import 'slick-carousel/slick/slick-theme.css'
import styles from '../ActiveRoutes/ActiveRoutes.scss'

export function Carousel ({ slides, noSlides, fade }) {
  if (!slides) {
    return ''
  }
  const slideNumber = noSlides || 4
  const settings = {
    dots: true,
    autoplay: true,
    infinite: true,
    slidesToShow: slideNumber,
    speed: 2000,
    arrows: false
  }
  const slideCountStyle = noSlides === 1 ? styles.one_slide : ''
  const StyledSlider = styled.div`
    .slick_slide {
      width: 1200px !important;
    }
  `

  const renderedSlides = slides.map((route) => {
    const divStyle = {
      backgroundImage: `url(${route.image})`
    }
    const slickSlide =
      navigator.userAgent.indexOf('MSIE') !== -1 || !!document.documentMode === true
        ? styles.slick_slide_ie11
        : styles.slick_slide
    return (
      <div
        key={v4()}
        className={`${slickSlide} flex-none layout-row layout-align-center-center`}
        style={divStyle}
      >
        {fade ? <div className={`flex-none ${styles.fade}`} /> : ''}
        <div
          className={`flex-none layout-column layout-align-center-center ${styles.slick_content}`}
        >
          <h2 className={`${styles.slick_city} flex-none`}> {route.header} </h2>
          <h5 className={`${styles.slick_country} flex-none`}> {route.subheader} </h5>
        </div>
      </div>
    )
  })
  return (
    <StyledSlider className={`flex-100 layout-row ${styles.slider_container} ${slideCountStyle}`}>
      <Slider {...settings}>{renderedSlides}</Slider>
    </StyledSlider>
  )
}

Carousel.propTypes = {
  slides: PropTypes.arrayOf(PropTypes.shape({
    image: PropTypes.string,
    header: PropTypes.string,
    subheader: PropTypes.string
  })).isRequired,
  noSlides: PropTypes.number,
  fade: PropTypes.bool
}

Carousel.defaultProps = {
  noSlides: 4,
  fade: false
}

export default Carousel
