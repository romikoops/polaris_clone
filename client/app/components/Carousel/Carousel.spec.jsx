import * as React from 'react'
import { shallow } from 'enzyme'

jest.mock('react-slick', () =>
  // eslint-disable-next-line react/prop-types
  ({ children }) => <span>{children}</span>)
jest.mock('node-uuid', () => ({
  v4: () => 'RANDOM_KEY'
}))
// eslint-disable-next-line
import { Carousel } from './Carousel'

const slideFirst = {
  image: 'FOO_IMAGE1',
  header: 'FOO_HEADER1',
  subheader: 'FOO_SUBHEADER1'
}
const slideSecond = {
  image: 'FOO_IMAGE2',
  header: 'FOO_HEADER2',
  subheader: 'FOO_SUBHEADER2'
}

const propsBase = {
  slides: [
    slideFirst,
    slideSecond
  ],
  noSlides: 1,
  fade: false
}

test('shallow render', () => {
  expect(shallow(<Carousel {...propsBase} />)).toMatchSnapshot()
})

test('props.fade is true', () => {
  const props = {
    ...propsBase,
    fade: true
  }
  expect(shallow(<Carousel {...props} />)).toMatchSnapshot()
})
