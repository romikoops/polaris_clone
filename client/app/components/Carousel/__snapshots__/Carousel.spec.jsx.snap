// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props.fade is true 1`] = `
<styled.div
  className="flex-100 layout-row slider_container one_slide"
>
  <Component
    arrows={false}
    autoplay={true}
    dots={true}
    infinite={true}
    slidesToShow={1}
    speed={2000}
  >
    <div
      className="slick_slide flex-none layout-row layout-align-center-center"
      key="RANDOM_KEY"
    >
      <div
        className="inner_wrapper flex-100 layout-row layout-align-center-center"
        style={
          Object {
            "backgroundImage": "url(FOO_IMAGE1)",
          }
        }
      >
        <div
          className="flex-none fade"
        />
        <div
          className="flex-none layout-column layout-align-center-center slick_content"
        >
          <h2
            className="slick_city flex-none"
          >
             
            FOO_HEADER1
             
          </h2>
          <h5
            className="slick_country flex-none"
          >
             
            FOO_SUBHEADER1
             
          </h5>
        </div>
      </div>
    </div>
    <div
      className="slick_slide flex-none layout-row layout-align-center-center"
      key="RANDOM_KEY"
    >
      <div
        className="inner_wrapper flex-100 layout-row layout-align-center-center"
        style={
          Object {
            "backgroundImage": "url(FOO_IMAGE2)",
          }
        }
      >
        <div
          className="flex-none fade"
        />
        <div
          className="flex-none layout-column layout-align-center-center slick_content"
        >
          <h2
            className="slick_city flex-none"
          >
             
            FOO_HEADER2
             
          </h2>
          <h5
            className="slick_country flex-none"
          >
             
            FOO_SUBHEADER2
             
          </h5>
        </div>
      </div>
    </div>
  </Component>
</styled.div>
`;

exports[`props.slides is falsy 1`] = `""`;

exports[`shallow render 1`] = `
<styled.div
  className="flex-100 layout-row slider_container one_slide"
>
  <Component
    arrows={false}
    autoplay={true}
    dots={true}
    infinite={true}
    slidesToShow={1}
    speed={2000}
  >
    <div
      className="slick_slide flex-none layout-row layout-align-center-center"
      key="RANDOM_KEY"
    >
      <div
        className="inner_wrapper flex-100 layout-row layout-align-center-center"
        style={
          Object {
            "backgroundImage": "url(FOO_IMAGE1)",
          }
        }
      >
        <div
          className="flex-none layout-column layout-align-center-center slick_content"
        >
          <h2
            className="slick_city flex-none"
          >
             
            FOO_HEADER1
             
          </h2>
          <h5
            className="slick_country flex-none"
          >
             
            FOO_SUBHEADER1
             
          </h5>
        </div>
      </div>
    </div>
    <div
      className="slick_slide flex-none layout-row layout-align-center-center"
      key="RANDOM_KEY"
    >
      <div
        className="inner_wrapper flex-100 layout-row layout-align-center-center"
        style={
          Object {
            "backgroundImage": "url(FOO_IMAGE2)",
          }
        }
      >
        <div
          className="flex-none layout-column layout-align-center-center slick_content"
        >
          <h2
            className="slick_city flex-none"
          >
             
            FOO_HEADER2
             
          </h2>
          <h5
            className="slick_country flex-none"
          >
             
            FOO_SUBHEADER2
             
          </h5>
        </div>
      </div>
    </div>
  </Component>
</styled.div>
`;
